### ISSUE TITLE
react/display-name false positive on ant-design ColumnsType

### ISSUE THREAD

**Issue by gaara4896** [eslint/eslint#13555]
**Tell us about your environment**

* **ESLint Version:** 7.5.0
* **Node Version:** 10.15.3
* **npm Version:** 6.4.1

**What parser (default, Babel-ESLint, etc.) are you using?**

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
  extends: [
    'eslint:recommended',
    'plugin:@typescript-eslint/eslint-recommended',
    'plugin:@typescript-eslint/recommended',
    'plugin:react/recommended',
    'plugin:jsx-a11y/recommended',
    // Prettier plugin and recommended rules
    'prettier/@typescript-eslint',
    'plugin:prettier/recommended',
  ],
  rules: {
    // Include .prettierrc.js rules
    'prettier/prettier': ['error', {}, { usePrettierrc: true }],
    'react/prop-types': 'error',
    'react/react-in-jsx-scope': 'error',
    '@typescript-eslint/explicit-function-return-type': 'error',
    '@typescript-eslint/ban-ts-ignore': 'off',
    'jsx-a11y/label-has-associated-control': [
      'error',
      {
        labelComponents: [],
        labelAttributes: [],
        controlComponents: [],
        assert: 'either',
        depth: 25,
      },
    ],
    'jsx-a11y/anchor-is-valid': 'off',
    '@typescript-eslint/no-explicit-any': 'off',
    '@typescript-eslint/explicit-module-boundary-types': 'error',
    '@typescript-eslint/no-unused-vars': 'error',
  },
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
...
import { ColumnsType } from "antd/lib/table";
import { Table } from "antd";

const profilesColumns: ColumnsType<any> = [
  {
    title: "ID",
    dataIndex: "id",
    key: "id"
  },
  {
    title: "URL",
    dataIndex: "url",
    key: "url",
    render: (url: string): React.ReactElement => (
      <a href={url} target="_blank" rel="noopener noreferrer">
        <p>lol</p>
      </a>
    )
  }
];

ReactDOM.render(
  <Table columns={profilesColumns} />,
  document.getElementById("root")
);
```

```bash
eslint --fix . --ext .ts,.tsx,.js,.jsx
```

**What did you expect to happen?**

No error.

**What actually happened? Please include the actual, raw output from ESLint.**

```bash
yarn run v1.17.3
$ eslint --fix . --ext .ts,.tsx,.js,.jsx

/home/user/my-project/components/ComponentTable.tsx
  32:13  error  Component definition is missing display name  react/display-name

✖ 1 problem (1 error, 0 warnings)

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Are you willing to submit a pull request to fix this bug?**
not really

**Comment by anikethsaha** [comment]
Thanks for the report. 

we don't maintain the `eslint-plugin-react` package, you can submit an issue in their [repo](https://github.com/yannickcr/eslint-plugin-react)

And as per the rule, it does expect a `DisplayName` as a property for the component as mention in the [docs](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/display-name.md)

**Comment by gaara4896** [comment]
For the Ant Design, `render` in `ColumnsType` is not a `React component` but function, hence the rule should not be applied in this scenario. 

Also thanks for the head up, I will open an issue in their repo

**Comment by kaicataldo** [comment]
Closing this issue since this project isn't maintained by this team.
